game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/ClearGeneralCasualties.java

```diff
@@ -0,0 +1,35 @@
+package games.strategy.triplea.delegate.battle.steps.change;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_CASUALTIES;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.List;
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class ClearGeneralCasualties implements BattleStep {
+  private static final long serialVersionUID = 995118832242624142L;
+
+  private final BattleState battleState;
+
+  private final BattleActions battleActions;
+
+  @Override
+  public List<String> getNames() {
+    return List.of(REMOVE_CASUALTIES);
+  }
+
+  @Override
+  public Order getOrder() {
+    return Order.GENERAL_REMOVE_CASUALTIES;
+  }
+
+  @Override
+  public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+    battleActions.clearWaitingToDieAndDamagedChangesInto(bridge);
```

Comment:
My first comment on this seems to have been lost in github : /

I'm wondering/concerned if this class is too thin and a bit indirect. Can we move the logic of `MustFightBattle` here and have the other references in `MustFightBattle` and elsewhere call this class? Even if we move that logic here as a static method, at least it would have the one, isolated home and presumably could be independently tested. Of most significant though we'd be unburdening `MustfightBattle` that is already doing too much.

Reply:
Yes, I plan to look into doing something like that now that most of the battle steps are converted.  You'll start seeing PR where I'm changing the `BattleAction` and `BattleState` apis and pull more of the code into the battlesteps or some sort of helper.

Reply:
:+1: